src/modules/collection/utils.ts

```diff
@@ -136,8 +138,12 @@ export function getCollectionFactoryContract(chainId: ChainId) {
   return getContract(ContractName.CollectionFactoryV3, chainId)
 }
 
-export function getRaritiesContract(chainId: ChainId) {
-  return env.get<string | undefined>('REACT_APP_FF_RARITIES_WITH_ORACLE') === '1'
-    ? getContract(ContractName.RaritiesWithOracle, chainId)
-    : getContract(ContractName.Rarities, chainId)
+export function* getRaritiesContract(chainId: ChainId) {
+  const enabled: boolean = yield select(getIsRaritiesWithOracleEnabled)
+
+  if (enabled) {
+    return getContract(ContractName.RaritiesWithOracle, chainId)
+  }
+
+  return getContract(ContractName.Rarities, chainId)
```

Comment:
What if instead of being a generator, it is a selector? It's used from a sagas, so we could use the `select` effect there and make this simpler.
It should be moved to selectors too.

```suggestion
export function getRaritiesContract(state: RootState, chainId: ChainId) {
  const enabled: boolean = getIsRaritiesWithOracleEnabled(state)

  if (enabled) {
    return getContract(ContractName.RaritiesWithOracle, chainId)
  }

  return getContract(ContractName.Rarities, chainId)
```

Reply:
I think that is a good idea. Had my doubts about adding a selector that returns something that does not belong directly to the store but there is actually no rule against it

Reply:
That is a great question, that seem to be a gray area. What made me recommend this change is that I tend to avoid whenever it's possible, `selects` in generators that are not action handlers. This is because to the problem of these types of function possibly being re-used with un-expected side-effects due to doing a select in a different moment in time where the state might have changed.

Reply:
Moved it